ansible/roles/collect_openshift_must_gather/tasks/main.yml

```diff
@@ -0,0 +1,41 @@
+---
+- name: Write kubeconfig if not exists and have api token and url
+  when:
+  - collect_openshift_must_gather_kubeconfig_path is not exists
+  - collect_openshift_must_gather_api_token != ''
+  - collect_openshift_must_gather_api_url != ''
+  ansible.builtin.copy:
+    dest: "{{ collect_openshift_must_gather_kubeconfig_path }}"
+    content: |
+      ---
+      apiVersion: v1
+      kind: Config
+      clusters:
+      - name: default
+        cluster:
+          server: {{ collect_openshift_must_gather_api_url | to_json }}
+      {% if collect_openshift_must_gather_api_ca_cert | default('') != '' %}
+          certificate-authority-data: {{ collect_openshift_must_gather_api_ca_cert | b64encode }}
+      {% endif %}
+      contexts:
+      - name: default
+        context:
+          cluster: default
+          namespace: default
+          user: default
+      current-context: default
+      preferences: {}
+      users:
+      - name: default
+        user:
+          token: {{ collect_openshift_must_gather_api_token | to_json }}
+
+- name: Run oc adm must-gather
+  ansible.builtin.command: >-
+    oc adm must-gather
+    --dest-dir="{{ output_dir }}/oc-adm-must-gather"
+    {% if collect_openshift_must_gather_kubeconfig_path is exists %}
+    --kubeconfig={{ collect_openshift_must_gather_kubeconfig_path }}
+    {% endif %}
+  ignore_errors: true
+...
```

Comment:
Accessing the must-gather post-destroy via s3 bucket might be challenging.  Just speculating: Is there any automation for investigators to grab it?  Maybe there's an even better place for them.

Reply:
Perhaps put a note here about how it fits into the larger solution.

Reply:
It would be good in the future to have a way to download the output dir from the RHDP ui. We don't have it on the roadmap yet, but we should add it.